x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/actions_log_with_rule_toggle.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFilterButton } from '@elastic/eui';
+import React, { useCallback } from 'react';
+import { useIsExperimentalFeatureEnabled } from '../../../../common/hooks/use_experimental_features';
+import { useActionHistoryUrlParams } from './use_action_history_url_params';
+import { FILTER_NAMES } from '../translations';
+
+interface ActionsLogWithRuleToggleProps {
+  isFlyout: boolean;
+  dataTestSubj?: string;
+}
+
+const ActionsLogWithRuleToggleContent = ({
+  isFlyout,
+  dataTestSubj,
+}: ActionsLogWithRuleToggleProps) => {
+  const responseActionsEnabled = useIsExperimentalFeatureEnabled('endpointResponseActionsEnabled');
+
+  const { withRuleActions: withRuleActionsUrlParam, setUrlWithRuleActions } =
+    useActionHistoryUrlParams();
+
+  const onClick = useCallback(() => {
+    if (!isFlyout) {
+      // set and show `withRuleActions` URL param on history page
+      setUrlWithRuleActions(!withRuleActionsUrlParam);
+    }
+  }, [isFlyout, setUrlWithRuleActions, withRuleActionsUrlParam]);
+
+  if (!responseActionsEnabled) return null;
+
+  return (
+    <EuiFilterButton
+      hasActiveFilters={withRuleActionsUrlParam}
+      onClick={onClick}
+      data-test-subj={`${dataTestSubj}-automated-responses-filter`}
+    >
+      {FILTER_NAMES.automated}
+    </EuiFilterButton>
```

Comment:
I feel this would be better with a Switch component instead. You can't really discern if the filter button is in an on or off state when toggled.

Reply:
We followed design guidelines followed by [this example](https://elastic.github.io/eui/#/tabular-content/in-memory-tables#in-memory-table-with-search-and-external-state).